src/main.cpp

```diff
@@ -540,21 +536,23 @@ unique_ptr<GbdxmArgs> readPackArgs(const po::variables_map& vm)
 
     // --color-mode
     if(vm.count("color-mode")) {
-        metadata.setColorMode(classification::colorModeFromString(vm["color-mode"].as<string>()));
-        DG_CHECK(metadata.colorMode() != classification::ColorMode::UNKNOWN,
-                "Invalid --color-mode argument");
-
-        tryErase(missingFields, "colorMode");
+        auto colorMode = classification::colorModeFromString(vm["color-mode"].as<string>());
+        if(colorMode != classification::ColorMode::UNKNOWN) {
+            metadata.setColorMode(colorMode);
+            tryErase(missingFields, "colorMode");
+        } else {
+            errors.emplace_back("Unsupported option for --color-mode '" + vm["color-mode"].as<string>() + "'");
+        }
     }
 
     // --resolution
     if(vm.count("resolution")) {
         metadata.setResolution(vm["resolution"].as<cv::Size2d>());
+        tryErase(missingFields, "resolution");
```

Comment:
Resolution is not mandatory, why do you still expect it to be in `missingFields`?

Reply:
I wanted to keep the logic of what is mandatory and what is optional in one place (`ModelMetadataJson::fieldNames(...)`) instead of defined there, in this main, and `ModelMetadataJson::fromJsonBase(...)`, having it in three places led to #31.